vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp

```diff
@@ -412,357 +413,541 @@ static void StyleYogaNode(
     const YGNodeRef yogaNode,
     const winrt::Microsoft::ReactNative::JSValueObject &props) {
   for (const auto &pair : props) {
-    const std::string &key = pair.first;
+    const auto &key = pair.first;
+    const auto key_hash = details::hash(pair.first.c_str());
     const auto &value = pair.second;
+    const auto value_hash = value.Type() == winrt::Microsoft::ReactNative::JSValueType::String
+        ? details::hash(value.TryGetString()->c_str())
+        : 0ull;
+
+    switch (key_hash) {
+      case HASH_STRING("flexDirection"): {
+        YGFlexDirection direction;
+
+        switch (value_hash) {
+          case HASH_STRING("column"):
+            direction = YGFlexDirectionColumn;
+            break;
+          case HASH_STRING("row"):
+            direction = YGFlexDirectionRow;
+            break;
+          case HASH_STRING("column-reverse"):
+            direction = YGFlexDirectionColumnReverse;
+            break;
+          case HASH_STRING("row-reverse"):
+            direction = YGFlexDirectionRowReverse;
+            break;
+          default:
+            direction = YGFlexDirectionColumn;
+            assert(value.IsNull());
+        }
+        YGNodeStyleSetFlexDirection(yogaNode, direction);
+        break;
+      }
+      case HASH_STRING("justifyContent"): {
+        YGJustify justify;
+
+        switch (value_hash) {
+          case HASH_STRING("flex-start"):
+            justify = YGJustifyFlexStart;
+            break;
+          case HASH_STRING("flex-end"):
+            justify = YGJustifyFlexEnd;
+            break;
+          case HASH_STRING("center"):
+            justify = YGJustifyCenter;
+            break;
+          case HASH_STRING("space-between"):
+            justify = YGJustifySpaceBetween;
+            break;
+          case HASH_STRING("space-around"):
+            justify = YGJustifySpaceAround;
+            break;
+          case HASH_STRING("space-evenly"):
+            justify = YGJustifySpaceEvenly;
+            break;
+          default:
+            justify = YGJustifyFlexStart;
+            assert(value.IsNull());
+        }
+        YGNodeStyleSetJustifyContent(yogaNode, justify);
+        break;
+      }
+      case HASH_STRING("flexWrap"): {
+        YGWrap wrap;
+
+        switch (value_hash) {
+          case HASH_STRING("nowrap"):
+            wrap = YGWrapNoWrap;
+            break;
+          case HASH_STRING("wrap"):
+            wrap = YGWrapWrap;
+            break;
+          case HASH_STRING("wrap-reverse"):
+            wrap = YGWrapWrapReverse;
+            break;
+          default:
+            wrap = YGWrapNoWrap;
+            assert(value.IsNull());
+
+            break;
+        }
+        YGNodeStyleSetFlexWrap(yogaNode, wrap);
+        break;
+      }
+      case HASH_STRING("alignItems"): {
+        YGAlign align;
+
+        switch (value_hash) {
```

Comment:
>switch [](http://example.com/codeflow?start=8&length=6)

How can we be sure that we always get a unique hash for the given set of string values?

Why not just create a std::unordered_map to achieve the same results? It will take care about any possible hash duplication.